### PR TITLE
Feature/dl 35 dl 38

### PR DESCRIPTION
**Pull request by marta789**
dashboard with borrowed booked and magazines added,
tests for results of queries added

### CODE REVIEW

File: static/css/dashboard.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#reserved_table{
+    display: block;
+}
+
+#booked_table{
+    display: none;
+}
+
+#reserv_reserv{
```
Comment (oskam) [inline static/css/dashboard.css]: Is this styling somewhere used?

File: templates/book_borrowing_dashboard.html [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                        <td>------</td>
+                        <td>------</td>
+                        <td>------</td>
+                    </tr>
+                  {% else %}
+                  {% for item in reserved_books %}
+                    <tr>
+                        <td>
+                            {{item[0].title}}
+                            {% if item[0].__tablename__ == "magazines" %}
```
Comment (oskam) [inline templates/book_borrowing_dashboard.html]: You could also just do here something like that: `{% if item.type == 'magazine' %}` - more readable.

File: static/js/book_borrowing_dashboard.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        document.getElementById("reserved_table").style.display = "block";
+        document.getElementById("reserv_button").style.textDecoration = "underline";
+        document.getElementById("booked_button").style.textDecoration = "none";
+    }
+    function booked_books() {
+        document.getElementById("reserved_table").style.display = "none";
+        document.getElementById("booked_table").style.display = "block";
+        document.getElementById("reserv_button").style.textDecoration = "none";
+        document.getElementById("booked_button").style.textDecoration = "underline";
+    }
```
Comment (tdiak) [inline static/js/book_borrowing_dashboard.js]: We are using Jquery in this project, so use: $("#myId").css('display', 'none') etc

File: static/js/book_borrowing_dashboard.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    window.onload = function() {
```
Comment (tdiak) [inline static/js/book_borrowing_dashboard.js]: use $(document).ready(function(){ ...})

File: templates/book_borrowing_dashboard.html [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                  <thead>
+                    <tr>
+                        <th scope="col">Title</th>
+                        <th scope="col">Author</th>
+                        <th scope="col">Reservation date</th>
+                        <th scope="col">Reservation expiry date</th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                  {% if reserved_books == [] %}
```
Comment (tdiak) [inline templates/book_borrowing_dashboard.html]: I think in Jinja2 empty list got a False value, so {% if reserved_books %} will be enought :)

File: templates/book_borrowing_dashboard.html (hunk 2 above)
Comment (tdiak) [reply templates/book_borrowing_dashboard.html]: Yes, item[0].__tablename__ looks ...weird :)

**Comment by oskam** [comment]
Change target of the pull request from `master` to `release/250618`.

File: tests/conftest.py [hunk 6]
```diff
@@ -251,3 +260,53 @@ def app_session(client, db_user):
...
+    authors = populate_authors(n=2)
+    session.add_all(authors)
+    session.commit()
+    books = populate_books(n=2, authors=authors)
+    session.add_all(books)
+    session.commit()
+    magazines = populate_magazines(n=2)
+    session.add_all(magazines)
+    session.commit()
+    copies = list()
```
Comment (etude-ist) [inline tests/conftest.py]: More pythonic way of handling that is to use just: `[]`

File: tests/conftest.py [hunk 7]
```diff
@@ -251,3 +260,53 @@ def app_session(client, db_user):
...
+    copies.append(populate_copies(books[0], n=1)[0])
+    copies.append(populate_copies(books[1], n=1)[0])
+    copies.append(populate_copies(magazines[0], n=1)[0])
+    copies.append(populate_copies(magazines[1], n=1)[0])
+    session.add_all(copies)
+    session.commit()
+    reservations = list()
```
Comment (etude-ist) [inline tests/conftest.py]: See previous comment.

File: views/book_borrowing_dashboard.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        filter(RentalLog.user_id == current_user_id). \
+        filter(RentalLog.copy_id == Copy.id). \
+        filter(LibraryItem.id == Copy.library_item_id). \
+        order_by(desc(RentalLog._borrow_time)). \
+        all()
+    return borrowed_items
+
+
+def get_number_of_reserved(s_db, current_user_id):
+    num_of_reserved = s_db.query(LibraryItem, RentalLog._reservation_begin,
```
Comment (etude-ist) [inline views/book_borrowing_dashboard.py]: I guess we could use previous query for getting that number. In that situation we do redundant query to db.

File: views/book_borrowing_dashboard.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        filter(RentalLog.book_status == BookStatus.RESERVED). \
+        filter(RentalLog.user_id == current_user_id). \
+        filter(RentalLog.copy_id == Copy.id). \
+        filter(LibraryItem.id == Copy.library_item_id). \
+        count()
+    return num_of_reserved
+
+
+def get_number_of_borrowed(s_db, current_user_id):
+    num_of_borrowed = s_db.query(LibraryItem, RentalLog._borrow_time,
```
Comment (etude-ist) [inline views/book_borrowing_dashboard.py]: Same as previous comment.

File: views/book_borrowing_dashboard.py (hunk 8 above)
Comment (marta789) [reply views/book_borrowing_dashboard.py]: I changed it for len(reserved_items) ,  can I somehow directly retrieve it from previous query

**Pull request merged**